File: lumera-crystal/frontend/lib/mail-command.ts
export type MailCommandResult = {
  status: "sent" | "ignored" | "error";
  message: string;
  mail?: {
    to: string;
    subject: string;
    body: string;
    cc: string[];
    bcc: string[];
    attachments: string[];
    raw_input: string;
    parsed_by: "rule" | "llm" | "hybrid";
    require_confirm: boolean;
  } | null;
  error?: {
    code: string;
    message: string;
  } | null;
};

const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL ?? "http://localhost:8000/api/v1";

export async function sendMailCommand(text: string): Promise<MailCommandResult> {
  const response = await fetch(`${API_BASE_URL}/ai/mail/command`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ text }),
  });

  if (!response.ok) {
    const raw = await response.text();
    let detail: unknown = raw;
    try {
      const parsed = raw ? JSON.parse(raw) : null;
      detail = parsed && typeof parsed === "object" && "detail" in parsed ? parsed.detail : parsed;
    } catch {
      detail = raw;
    }
    const message = typeof detail === "string" ? detail : "请求失败";
    throw new Error(message);
  }

  return response.json() as Promise<MailCommandResult>;
}

File: lumera-crystal/backend/app/schemas/mail_command.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class MailCommandInput(BaseModel):
    text: str = Field(..., min_length=1, description="Raw user input")


class MailCommand(BaseModel):
    to: str
    subject: str
    body: str
    cc: list[str] = Field(default_factory=list)
    bcc: list[str] = Field(default_factory=list)
    attachments: list[str] = Field(default_factory=list)
    raw_input: str
    parsed_by: Literal["rule", "llm", "hybrid"]
    require_confirm: bool = False


class MailCommandError(BaseModel):
    code: str
    message: str


class MailCommandResult(BaseModel):
    status: Literal["sent", "ignored", "error"]
    message: str
    mail: MailCommand | None = None
    error: MailCommandError | None = None
